### PR TITLE
High speed VTOL flight fixes

### PR DESCRIPTION
**Pull request by priseborough**
This fixes a bug that was incorrectly applying the forward flight motor failure detection to tilt rotors.

This adds a protection to prevent quadplanes from performing excessive pitch up during braking manoeuvres in VTOL modes that could over stress the airframe.

The following figure is from a simulation test and shows how the upper pitch angle limit lowers as the vehicle speeds up in a forward acceleration followed by a braking manoeuvre in QLOITER.

Screenshot 2024-02-20 at 9 23 34 am

We will update as testing continues and we determine the appropriate default value for the Q_BCK_PIT_LIM parameter. The limit can be disabled by setting the parameter to a non positive value.

This is something that should be considered for the Beta release.

### CODE REVIEW

File: ArduPlane/quadplane.cpp [hunk 1]
```diff
@@ -529,6 +529,15 @@ const AP_Param::GroupInfo QuadPlane::var_info2[] = {
     // @User: Standard
     AP_GROUPINFO("FWD_THR_USE", 37, QuadPlane, q_fwd_thr_use, uint8_t(FwdThrUse::OFF)),
 
+    // @Param: BCK_PIT_LIM
+    // @DisplayName: Q mode rearward pitch limit
+    // @Description: This sets the maximum number of degrees of back or pitch up in Q modes when the airspeed is at AIRSPEED_FBW_MIN and is used to prevent excessive sutructural loads when pitching up decelerate. If airspeed is above or below AIRSPEED_FBW_MIN the pitch up/back will be adjusted according to the formula pitch_limit = BCK_PIT_LIM * (AIRSPEED_FBW_MIN / IAS)^2. The backwards/up pitch limit controlled by this parameter is in addition to limiting applied by PTCH_LIM_MAX and Q_ANGLE_MAX. The BCK_PIT_LIM limit is only applied when Q_FWD_THR_USE is set to 1 or 2 and the vehicle is flying in a mode that uses forward throttle instead of forward tilt to generate forward speed. Set to a non positive value 0 to deactivate this limit.
```
Comment (tridge) [inline ArduPlane/quadplane.cpp:534]: note that AIRSPEED_FBW_MIN is now called AIRSPEED_MIN

**Review by tridge — CHANGES_REQUESTED** [review]
minor spelling error ...
has this been flight tested? It looks like something we would want in 4.5.0

**Comment by priseborough** [comment]
> > has this been flight tested? It looks like something we would want in 4.5.0
I'm trying to get a cygwin binary to a tester that has a representative model of their quadplane before they go and fly it on hardware. The artifacts produced by the build server don't contain the cygwin binaries I was expecting.

https://github.com/priseborough/ardupilot/actions/runs/7892856368

**Comment by tridge** [comment]
> The artifacts produced by the build server don't contain the cygwin binaries I was expecting.

thanks, fix here:
https://github.com/ArduPilot/ardupilot/pull/26215
maybe cherry-pick that into this PR?

**Comment by tridge** [comment]
@priseborough I've rebased this on the cygwin fix

**Comment by pompecukor** [comment]
Again thank you, we been collaborating with Paul on this and very happy for the solution. Tested in SITL and we are seeing what we wanted. Looking forward for it in 4.5 as it is a safety "must" for the new forward VTOL speed PR.

**Comment by priseborough** [comment]
Here are the results from the last round of simulation testing with Q_BCK_PIT_LIM reduced fro 10 to 5 degrees so that the pitch is being limited during high speed braking manoeuvres. This test was into zero wind. During the decel, the nav pitch upper pitch limit FWDT.npulcd is limiting the pitch angle during the intial phase of trhe braking maneouvre. 

Screenshot 2024-02-21 at 7 22 31 am

The velocity loop handles the limiting, noting that during decel at higher airspeeds the deceleration achieved has a lot of rotor momentum and other drag contributing to it which the position controller ignores with its assumption that the only forces are rotor thrust and gravity.

Screenshot 2024-02-21 at 7 23 13 am
Screenshot 2024-02-21 at 7 23 05 am

The same manoeuvre was also tested  into a high wind speed:

Screenshot 2024-02-21 at 7 18 02 am

The duration of pitch limiting is longer. Note that the vehicle gets blown back a few metres at the end of the braking phase before stopping. This pre-exisiting characteristic.

Screenshot 2024-02-21 at 7 18 11 am

Screenshot 2024-02-21 at 7 18 21 am

We are now at a point were flight testing is required.

**Comment by tridge** [comment]
@priseborough looks good!

File: ArduPlane/quadplane.cpp [hunk 2]
```diff
@@ -529,6 +529,15 @@ const AP_Param::GroupInfo QuadPlane::var_info2[] = {
     // @User: Standard
     AP_GROUPINFO("FWD_THR_USE", 37, QuadPlane, q_fwd_thr_use, uint8_t(FwdThrUse::OFF)),
 
+    // @Param: BCK_PIT_LIM
+    // @DisplayName: Q mode rearward pitch limit
+    // @Description: This sets the maximum number of degrees of back or pitch up in Q modes when the airspeed is at AIRSPEED_MIN and is used to prevent excessive sutructural loads when pitching up decelerate. If airspeed is above or below AIRSPEED_FBW_MIN the pitch up/back will be adjusted according to the formula pitch_limit = BCK_PIT_LIM * (AIRSPEED_FBW_MIN / IAS)^2. The backwards/up pitch limit controlled by this parameter is in addition to limiting applied by PTCH_LIM_MAX and Q_ANGLE_MAX. The BCK_PIT_LIM limit is only applied when Q_FWD_THR_USE is set to 1 or 2 and the vehicle is flying in a mode that uses forward throttle instead of forward tilt to generate forward speed. Set to a non positive value 0 to deactivate this limit.
```
Comment (tridge) [inline ArduPlane/quadplane.cpp:534]: need to replace AIRSPEED_FBW_MIN with AIRSPEED_MIN in the description in 2 places, also PTCH_LIM_MAX
is now PTCH_LIM_MAX_DEG

**Review by tridge — CHANGES_REQUESTED** [review]

**Comment by pompecukor** [comment]
Now flight tested on real craft @priseborough and @tridge have the logs

**Review by tridge — APPROVED** [review]

**Comment by priseborough** [comment]
Here are the flight test results from a high QLOITER speed dash into wind with a braking manoeuvre. The up pitch is limited in part of the decel manoeuvre using Q_BCK_PIT_LIM = 5. I would rather leave the value at 10 by default and leave it o users to reduce the value if they think the braking manoeuvre is overloading the wing rather than have a value that could be excessively restrictive and reduce braking performance for other users.

Screenshot 2024-02-24 at 3 39 16 pm

Screenshot 2024-02-24 at 3 39 26 pm

**Pull request merged**